api/src/test/java/org/openmrs/api/PatientServiceTest.java

```diff
@@ -1881,7 +1934,7 @@ public void getAllPatients_shouldFetchVoidedPatientsWhenGivenIncludeVoidedIsTrue
 		List<Patient> allPatients = patientService.getAllPatients(true);
 		// there are 1 voided and 4 nonvoided patients in
 		// standardTestDataset.xml
-		assertEquals(6, allPatients.size());
+		assertEquals(7, allPatients.size());
```

Comment:
Why is this necessary?

Reply:
Hi, <br/>I have been facing this challenge for a while. Its all about Threads saving the data to the "in-memory-database" and effects aren't reverted as the default behavior is to other tests. Could I be missing something when using Threads?   
The above happens because of this [line](https://github.com/samuelmale/openmrs-core/blob/ce0953e35f0ddfaff07c738156cbcac386ab82af/api/src/test/java/org/openmrs/api/PatientServiceTest.java#L1876). Meaning the patient is saved the effect ain't reverted hence I had to change the number of patients from 6 to 7.  Its exactly the same issue in this thread. Could you please advise me on this?

@dkayiwa also asked me about it [here](https://github.com/openmrs/openmrs-core/pull/2562#discussion_r177600261).

Reply:
All I know is that changes in one thread aren't visible in others in tests since the transactions get rollback.

What I don't understand is why it's happening now and not before because the test was passing before your changes